src/masternode.cpp

```diff
@@ -140,22 +140,6 @@ CMasternode::CollateralStatus CMasternode::CheckCollateral(const COutPoint& outp
         return COLLATERAL_INVALID_PUBKEY;
     }
 
-    CTransactionRef tx;
-    uint256 hashBlock;
-    if (!GetTransaction(outpoint.hash, tx, Params().GetConsensus(), hashBlock, true)) {
-        // should not happen
-        return COLLATERAL_UTXO_NOT_FOUND;
-    }
```

Comment:
Why should this part be removed?

Reply:
This was a duplicate check which I initially introduced as part of the "no old collaterals" rules. See the first check that happens in this method, it already checks for not existing or spent UTXOs.

The first check however did not give us the block hash, which I needed in the next check. And as I really don't like doing unchecked calls to anything, I prefer to be explicit and verbose about such things (see comment). Now we don't need the call at all, so we can remove it completely.